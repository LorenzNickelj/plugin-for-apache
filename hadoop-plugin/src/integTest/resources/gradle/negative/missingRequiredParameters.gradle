plugins {
  id 'com.linkedin.gradle.hadoop.HadoopPlugin'
}

// Negative test for missing required parameters.
hadoop {
  buildPath "jobs"
  cleanPath false

  workflow('missingRequiredParametersWorkflow1') {

    // The first job sets the correct required properties
    job('job1') {
      required parameters: [
          'a', 'b', 'c'
      ]
      set properties: [
          'a': 1,
          'b': 2,
          'c': 3
      ]
    }

    // The second does not set the required properties 'b' and 'c'
    job('job2') {
      required parameters: ['a', 'b', 'c']
      set properties: [
          'a': 1
      ]
    }

    targets 'job1', 'job2'
  }
}
