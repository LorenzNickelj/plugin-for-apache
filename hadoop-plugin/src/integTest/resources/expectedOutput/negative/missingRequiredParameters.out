RequiredParametersChecker ERROR: Job missingRequiredParametersWorkflow1/job2 requires parameter b, but it is not set.
RequiredParametersChecker ERROR: Job missingRequiredParametersWorkflow1/job2 requires parameter c, but it is not set.
